### ISSUE TITLE
Generalized IOU loss doesn't handle 0-length and -width boxes correctly, expected 0.0 but got 1.0

### ISSUE THREAD

**Issue by Maninae** [pytorch/vision#7746]
### 🐛 Describe the bug

Generalized IOU loss doesn't handle boxes that are parametrized as 0 length and 0 width correctly. The expectation is that with identical boxes, the GIOU loss is close to zero. This is true for non-trivial boxes with positive length and width. For trivial boxes [0, 0, 0, 0] compared against itself, the GIOU loss returns `1.` where we expected the loss function to return `0.` This is an edge case.

In the [implementation](https://pytorch.org/vision/main/_modules/torchvision/ops/giou_loss.html#generalized_box_iou_loss), this is because `iouk` is 0 when iou=0 and union=0 due to the epsilon in the denominator. Can we catch situations where iou == union == 0, and return 1?

Hoping this behavior can be caught and aligned, so that GIOU returns 0 for trivial boxes too.

```
>>> boxes = torch.tensor([[0, 0, 0, 0]])
>>> torchvision.ops.generalized_box_iou_loss(boxes, boxes)
tensor([1.])  # this should be 0
>>> boxes = torch.tensor([[0, 0, 1, 1]])
>>> torchvision.ops.generalized_box_iou_loss(boxes, boxes)
tensor([1.1921e-07])
```

### Versions

most recent version

**Comment by abhi-glitchhg** [comment]
The boxes are expected to be in `x1,y1,x2,y2` format where `0<=x1<x2` and `0<=y1<y2`. 

 [0,0,0,0] box violates the above condition.

**Comment by oke-aditya** [comment]
And we don't do any validations to avoid CUDA calls. So it's kind of expected that we throw abrupt value. 
Can you try with 

```
>>> boxes = torch.tensor([[0, 1e-5, 0, 1e-5]])
>>> torchvision.ops.generalized_box_iou_loss(boxes, boxes)
```

And see what is the Output? Is it close to 1?

**Comment by NicolasHug** [comment]
As @abhi-glitchhg pointed out, the bbox utils will generally not work properly on degenerate boxes. @Maninae  are you able to guard against such boxes in your code prior to calling `generalized_box_iou_loss()`?
